keras_tuner/engine/oracle.py

```diff
@@ -25,21 +24,23 @@
 
 from keras_tuner import utils
 from keras_tuner.engine import hyperparameters as hp_module
-from keras_tuner.engine import metrics_tracking
+from keras_tuner.engine import objective as obj_module
 from keras_tuner.engine import stateful
 from keras_tuner.engine import trial as trial_lib
 
-Objective = collections.namedtuple("Objective", "name direction")
-
 
 class Oracle(stateful.Stateful):
     """Implements a hyperparameter optimization algorithm.
 
     Args:
-        objective: A string or `keras_tuner.Objective` instance. If a string,
-            the direction of the optimization (min or max) will be inferred.
-            It is optional when `Tuner.run_trial()` or `HyperModel.fit()`
-            returns a single float as the objective to minimize.
+        objective: A string, `keras_tuner.Objective` instance, or a list of
+            `keras_tuner.Objective`s and strings. If a string, the direction of
+            the optimization (min or max) will be inferred. If a list of
+            `keras_tuner.Objective`, we will minimize the sum of all the
+            objectives to minimize subtracting the sum of all the objectives to
+            maximize. The `objective` argument is optional when
+            `Tuner.run_trial()` or `HyperModel.fit()` returns a single float as
+            the objective to minimize.
```

Comment:
Should we support giving different weights to the different metrics in the multi objective? If so, how should we design the API for passing the weights?

Reply:
It will be in a new PR.

Reply:
@haifeng-jin which PR did the weight support end up being in?

Reply:
I haven't started to work on it yet. The current work around might be more convenient. You can just override `HyperModel.fit()` and compute the weighted sum you want and return it.